python/summarize_sgfs.py

```diff
@@ -0,0 +1,295 @@
+      self._warning = True
+
+    if (pla_black, pla_white) not in self.results:
+      self.results[(pla_black, pla_white)] = Record()
+
+    if (winner == 'b'):
+      self.results[(pla_black, pla_white)].win += 1
+    elif (winner == 'w'):
+      self.results[(pla_black, pla_white)].lost += 1
+    else:
+      self.results[(pla_black, pla_white)].draw += 1
+
+  def _estimate_elo(self) -> elo.EloInfo:
+    """Estimate and print elo values. This function must be called after add all the sgfs/sgf files"""
+    pla_names = set(itertools.chain(*(name_pair for name_pair in self.results.keys())))
+    data = []
+    for pla_Balck in pla_names:
+      for pla_White in pla_names:
+        if (pla_Balck == pla_White):
+          continue
+        else:
+          record = self.results[(pla_Balck, pla_White)]
+          total = record.win + record.lost + record.draw
+          win = record.win + 0.5 * record.draw
+          winrate = win / total
+          data.extend(elo.likelihood_of_games(pla_Balck, pla_White, total, winrate, False))
+    data.extend(elo.make_center_elos_prior(list(pla_names), self._elo_prior))
+    info = elo.compute_elos(data, verbose=True)
+    return info
+
+  def _print_player_info(self):
+    pla_names = set(itertools.chain(*(name_pair for name_pair in self.results.keys())))
+    max_len = len(max(pla_names, key=len))
+    title = 'Player Name'
+    title_space = max(len(title), max_len) + 1
+    print(f"{title:<{title_space}}: Player ID")
+    idx = 0
+    for pla in pla_names:
+      print(f"{pla:<{title_space}}: {idx}")
+      idx += 1
+
+  def _print_result_matrix(self):
+    results_matrix = self._build_result_matrix()
+    max_game_played_space = int(math.log10(max([max(sublist) for sublist in results_matrix]))) + 5
+    row_format = f"{{:>{max_game_played_space}}}" * (len(results_matrix) + 1)
+    print(row_format.format("", *range(len(results_matrix))))
+    for playerID, row in zip(range(len(results_matrix)), results_matrix):
+      print(row_format.format(playerID, *row))
+
+  def _build_result_matrix(self) -> list:
+    pla_names = list(set(itertools.chain(*(name_pair for name_pair in self.results.keys()))))
+    results_matrix = []
+    for pla1 in pla_names:
+      row = []
+      for pla2 in pla_names:
+        if (pla1 == pla2):
+          row.append(0)
+          continue
+        else:
+          pla1_pla2 = self.results[(pla1, pla2)]
+          pla2_pla1 = self.results[(pla2, pla1)]
+          win = pla1_pla2.win + pla2_pla1.lost + 0.5 * (pla1_pla2.draw + pla2_pla1.draw)
+          row.append(win)
+      results_matrix.append(row)
+    return results_matrix
+
+
+if __name__ == "__main__":
+  description = """
+  Summarize SGF/SGFs files and estimate Bayes Elo score for each of the player.
+  """
+  parser = argparse.ArgumentParser(description=description)
+  parser.add_argument('-input-dirs', help='sgf/sgfs files input directories. If multiple directories, seperate them '
```

Comment:
Make this a positional argument, one that doesn't require a flag?

Reply:
save reason as above.

Reply:
This one I don't understand the reasoning. How does making this require a flag help against including SGF files? The only difference is whether the user has to type `-input-dirs` or not before typing the list of files.

Reply:
against including *mistaken* SGF files.

Reply:
I was thinking to be more specific. Let me know if you prefer other way, we can modify it. I am absolutely flexible here.